modules/doc/content/ncrc/applications/ncrc_develop.md.template

```diff
@@ -0,0 +1,154 @@
+# {{ApplicationName}} Development
+
+Proper development of {{ApplicationName}} requires a compiler, a clone of the {{ApplicationName}}
+repository, and any additional support libraries in the form of submodules. The following
+instructions aim at preparing your machine or environment for such development.
+
+If you encounter errors during any of these steps, you need to stop and figure out a solution before
+continuing. Please see our [Troubleshooting](help/troubleshooting.md) guide, which also contains
```

Comment:
```suggestion
continuing. Please see our [Troubleshooting](help/troubleshooting.md optional=True) guide, which also contains
```

This needs to be optional to allow sockeye documentation to cleanly build (along with some other [changes I have pushed up to a PR there](https://github.inl.gov/ncrc/sockeye/pull/635)).

Reply:
Gonna to try and convince you all that using an https link, instead of internal... to get around this crazy rabbit hole.

Reply:
I would not be against that idea....it's just fragile if we re-organize things and forget to change it.